apiary.apib

```diff
@@ -1682,6 +1682,364 @@ asynchronously once the job is created.
+
+### Retrieve existing order [GET]
+
++ Parameters
+    + factory_slug (string) - The unique identifier of required factory location
+    + order_id (string) - The unique ID of the required order
+
++ Request
+    + Headers
+
+            Authorization: Token ABCDEF
+
++ Response 200 (application/json)
+
+    + Body
+
+            {}
+
+## List Order Jobs [/v2/factories/{factory_slug}/orders/{order_id}/jobs{?state,page}]
+
+This endpoint allows you to retrieve manufacturing data for the individual jobs in a given order. Orders can have
+an arbitrary number of jobs, so this endpoint is paginated to show 100 jobs per page.
+
+### List jobs [GET]
+
++ Parameters
+    + factory_slug (string) - The unique identifier of required factory location
+    + order_id (string) - The unique ID of the required order
+    + state (optional, string) - Only show jobs with this state (see Job States section above for allowed values)
+    + page (optional, number) - Page number
+
++ Request
+    + Headers
+
+            Authorization: Token ABCDEF
+
++ Response 200 (application/json)
+
+    + Body
+
+            {}
+
+## List All Jobs [/v2/factories/{factory_slug}/jobs{?state,page}]
+
+If you prefer, you can also view a list of all jobs available in a given factory location. This shows jobs across
+all orders in this location.
+
+### List jobs [GET]
+
++ Parameters
+    + factory_slug (string) - The unique identifier of required factory location
+    + state (optional, string) - Only show jobs with this state (see Job States section above for allowed values)
+    + page (optional, number) - Page number
+
++ Request
+    + Headers
+
+            Authorization: Token ABCDEF
+
++ Response 200 (application/json)
+
+    + Body
+
+            {}
+
+## Retrieve Job [/v2/factories/{factory_slug}/jobs/{job_id}]
+
+The response from this endpoint contains the same data as the job representation in the list endpoints.
+
+### List jobs [GET]
+
++ Parameters
+    + factory_slug (string) - The unique identifier of required factory location
+    + job_id (string) - The unique identifier of specified job
+
++ Request
+    + Headers
+
+            Authorization: Token ABCDEF
+
++ Response 200 (application/json)
+
+    + Body
+
+            {}
+
+## Set Job In Production [/v2/factories/{factory_slug}/jobs/{job_id}/set_in_production]
+
+Mark this job as `in_production`. Jobs can only be marked `in_production` from the `new` state.
+
+If this job's order is the `new` state, it is automatically updated to the `in_production` state along with the given
+job.
+
+### Set Job In Production [POST]
+
++ Parameters
+    + factory_slug (string) - The unique identifier of required factory location
+    + job_id (string) - The unique identifier of specified job
+
++ Request
+    + Headers
+
+            Authorization: Token ABCDEF
+
++ Response 200 (application/json)
+
+    + Body
+
+            {}
+
+## Set Job Ready to Ship [/v2/factories/{factory_slug}/jobs/{job_id}/set_ready_to_ship]
```

Comment:
regarding updating states in general - do we wanna be more flexible, with a `force` type argument since the reality is that factories don't go always go through the order states in a linear fashion (this would obviously mean our state transitions are more flexible).
P.S. this might not be a discussion for this time

Reply:
Yes, totally. I think it's a problem in the UI too though right? I'd want to figure it out all at once

Reply:
it is a problem in the UI also, yes